adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m

```diff
@@ -0,0 +1,140 @@
+
+  if (!strongConnector || !strongAdapter) {
+    NSLog(@"Adaptor Error: No GADMAdNetworkConnector or GADMAdNetworkAdapter found for getBannerWithSize:");
+    return;
+  }
+
+  NSError *error = nil;
+  if (error) {
+    [strongConnector adapter:strongAdapter didFailAd:error];
+    return;
+  }
+
+  _placementID = [[[strongConnector credentials] objectForKey:kGADMAdapterUnityPlacementID] copy];
+
+  if (!_placementID){
+    NSString *description = [NSString
+                             stringWithFormat:@"Tried to initialize %@ with nil placement.", NSStringFromClass([UADSBannerView class])];
+    NSError *error = GADUnityErrorWithDescription(description);
+    [strongConnector adapter:strongAdapter didFailAd:error];
+    return;
+  }
+
+  adSize.size.width = adSize.size.width < 320 ? 320 : adSize.size.width;
+  adSize.size.height = adSize.size.height < 50 ? 50 : adSize.size.height;
+
+  _bannerAd = [[UADSBannerView alloc] initWithPlacementId:_placementID size:adSize.size];
+
+  if (!_bannerAd) {
+    NSString *description = [NSString
+                             stringWithFormat:@"%@ failed to initialize.", NSStringFromClass([UADSBannerView class])];
+    NSError *error = GADUnityErrorWithDescription(description);
+    [strongConnector adapter:strongAdapter didFailAd:error];
+    return;
+  }
+
+  [_bannerAd setDelegate:self];
+  [_bannerAd load];
+}
+
+- (void)stopBeingDelegate {
+  _bannerAd = nil;
+}
+
+#pragma mark UADSBannerView Delegate methods
+
+- (void)bannerViewDidLoad:(UADSBannerView *)bannerView {
+  id<GADMAdNetworkConnector> strongConnector = _connector;
+  id<GADMAdNetworkAdapter> strongAdapter = _adapter;
+
+  if (strongConnector && strongAdapter) {
+    [strongConnector adapter:strongAdapter didReceiveAdView:bannerView];
+  }
+}
+
+
+- (void)bannerViewDidClick:(UADSBannerView *)bannerView {
+  id<GADMAdNetworkConnector> strongConnector = _connector;
+  id<GADMAdNetworkAdapter> strongAdapter = _adapter;
+  if (strongAdapter && strongConnector) {
+    [strongConnector adapterDidDismissFullScreenModal:strongAdapter];
+  }
+}
+
+
+- (void)bannerViewDidLeaveApplication:(UADSBannerView *)bannerView {
+  id<GADMAdNetworkConnector> strongConnector = _connector;
+  id<GADMAdNetworkAdapter> strongAdapter = _adapter;
+  if (strongAdapter && strongConnector) {
+    [strongConnector adapterWillLeaveApplication:strongAdapter];
+  }
+}
+
+
+- (void)bannerViewDidError:(UADSBannerView *)bannerView error:(UADSBannerError *)error {
+  id<GADMAdNetworkConnector> strongConnector = _connector;
+  id<GADMAdNetworkAdapter> strongAdapter = _adapter;
+  if (strongConnector && strongAdapter) {
+    if ( error.code == UADSBannerErrorCodeNoFillError) {
+      [strongConnector adapter:strongAdapter didFailAd:GADUnityErrorWithDescription(@"Unity Ads banner failed to load - No Fill.")];
+    } else {
+      [strongConnector adapter:strongAdapter didFailAd:GADUnityErrorWithDescription(@"Unity Ads banner failed to load - Error.")];
+    }
```

Comment:
Better to make if statement only have the string, and to have 1 line of code to invoke the callback using the string. Less repetitive code.

Reply:
Fixed it.